### PR TITLE
✨implementation of amp-quicktext extension

### PR DESCRIPTION
**Pull request by MalekBouba**
This is the first PR for the new amp-quicktext extension.

### What does the extension do?

This extension will allow AMP developer to integrate and use the [Quicktext](https://www.quicktext.im/) live chat widget. Quicktext is a tool like Intercom but specialized in Hotels affairs and Hospitality, the widget is already in production and integrated into more than 1700 hotel's website in 5 continents.

### Why we build it?

Recently, we have some new customers who use AMP or old ones moving to it (and it looks like the number is growing), So, we decided to help them integrate our module using the easiest and the standard way.

### What this would look like?

Visually, the widget is a floating component, fully customizable with the host graphics (position, color, font...). It starts as a circular button, then it extends to show the chat container.
Technically, the widget is a React application encapsulated in iframes. it doesn't leak or interfere in any way with the host not in style, neither in events or layout.

### What's new?

- AMP Websites can easily with this extension integrate Quicktext widget.
- developers can customize the widget easily with the extension's attributes.

### CODE REVIEW

**Comment by MalekBouba** [comment]
@googlebot I signed it!

File: extensions/amp-quicktext/0.1/test/validator-amp-quicktext.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!doctype html>
```
Comment (Gregable) [inline extensions/amp-quicktext/0.1/test/validator-amp-quicktext.html]: Please add the AMP Project license to this file, and update the test output to match.

**Comment by nainar** [comment]
Hi @MalekBouba - what does the `<amp-quicktext>` extension do? Could you mention that in the PR description?

**Comment by sparhami** [comment]
This extension appears to mostly load some external JS rather than being implemented in AMP itself. AMP extensions are either implemented in AMP (following conventions like using `mutateElement` for DOM mutations, only doing changing the UI in response to user actions) or done using an iframe (e.g. amp-twitter).

Taking a step back, are the specific reasons why an extension is necessary (would it be possible for developers to access this functionality with just an amp-iframe)? Note that you can still do things like requesting a resize of the iframe if using one. Are there specific APIs that we could add that would enable this to be done with an iframe instead?

If users using amp-iframe is not an option, it seems like we would want to do a similar pattern to other extensions loading third-party (e.g. twitter, facebook), using an iframe to make sure the third party script does not run on the original domain. The twitter extension implementation is:

https://github.com/ampproject/amphtml/blob/master/extensions/amp-twitter/0.1/amp-twitter.js

It creates an iframe here:

https://github.com/ampproject/amphtml/blob/d1a7ee5a00f9e4cf08a06aeff321c97aa34c0517/extensions/amp-twitter/0.1/amp-twitter.js#L73-L75

Which loads this code in an iframe:

https://github.com/ampproject/amphtml/blob/master/3p/twitter.js

Which then loads some external javascript here:

https://github.com/ampproject/amphtml/blob/d1a7ee5a00f9e4cf08a06aeff321c97aa34c0517/3p/twitter.js#L32-L34

**Comment by MalekBouba** [comment]
> Hi @MalekBouba - what does the `<amp-quicktext>` extension do? Could you mention that in the PR description?
> 
> Also a visual demo of what this would look like and who it would be used by would be useful?

@nainar The description is updated. you can also check this [website](https://www.zafirohotels.com/) as an example to see the widget in action.
Thanks for your feedback.

**Comment by MalekBouba** [comment]
> This extension appears to mostly load some external JS rather than being implemented in AMP itself. AMP extensions are either implemented in AMP (following conventions like using `mutateElement` for DOM mutations, only doing changing the UI in response to user actions) or done using an iframe (e.g. amp-twitter).
> 
> Taking a step back, are the specific reasons why an extension is necessary (would it be possible for developers to access this functionality with just an amp-iframe)? Note that you can still do things like requesting a resize of the iframe if using one. Are there specific APIs that we could add that would enable this to be done with an iframe instead?
> 
> If users using amp-iframe is not an option, it seems like we would want to do a similar pattern to other extensions loading third-party (e.g. twitter, facebook), using an iframe to make sure the third party script does not run on the original domain. The twitter extension implementation is:
> 
> https://github.com/ampproject/amphtml/blob/master/extensions/amp-twitter/0.1/amp-twitter.js
> 
> It creates an iframe here:
> 
> https://github.com/ampproject/amphtml/blob/d1a7ee5a00f9e4cf08a06aeff321c97aa34c0517/extensions/amp-twitter/0.1/amp-twitter.js#L73-L75
> 
> Which loads this code in an iframe:
> 
> https://github.com/ampproject/amphtml/blob/master/3p/twitter.js
> 
> Which then loads some external javascript here:
> 
> https://github.com/ampproject/amphtml/blob/d1a7ee5a00f9e4cf08a06aeff321c97aa34c0517/3p/twitter.js#L32-L34

@sparhami  I'm afraid that wrapping this widget inside an iframe in that way is totally impossible for many reasons: style & UX, complexity, etc. nonetheless that it is already working in thousands of hotels with different technologies (integrated pages with JQuery, Gatsby, Angular, React...).
We tried also to use the <amp-script> before we settle on developing an extension.
On the other hand, we have done what it takes to prevent any leak from inside the widget (Cause this is a must when you run in a host website) and we also use MutationObservers to prevent any interference in the work or style of our widget.
Finally, we are ready to fix anything you see that can threaten the stability of our/your customer's website.

File: extensions/amp-quicktext/0.1/test/validator-amp-quicktext.html (hunk 1 above)
Comment (Gregable) [reply extensions/amp-quicktext/0.1/test/validator-amp-quicktext.html]: It looks like the license has been added, but the corresponding `.out` file does not show the license.

File: extensions/amp-quicktext/OWNERS [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// For an explanation of the OWNERS rules and syntax, see:
+// https://github.com/ampproject/amp-github-apps/blob/master/owners/OWNERS.example
+
+{
+  rules: [
+  {
+    owners: [
+      {name: 'satoripop/amp'},
```
Comment (rcebulko) [inline extensions/amp-quicktext/OWNERS]: Non-`ampproject` GitHub teams are not supported as owners at this time. All extensions should generally also have an `ampproject` working group owner, in this case probably `ampproject/wg-ui-and-a11y`

File: extensions/amp-quicktext/0.1/test/validator-amp-quicktext.html (hunk 1 above)
Comment (Gregable) [reply extensions/amp-quicktext/0.1/test/validator-amp-quicktext.html]: Is this PR still outstanding?

**Comment by mrjoro** [comment]
Closing as this PR no longer appears active.  Please reopen if needed.

**Pull request closed without merging**